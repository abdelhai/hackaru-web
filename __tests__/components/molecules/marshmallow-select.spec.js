import Factory from '@/__tests__/__setups__/factory';
import MarshmallowSelect from '@/components/molecules/marshmallow-select';

describe('MarshmallowSelect', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(MarshmallowSelect, {
      propsData: {
        ariaLabel: 'Fruits',
        value: 'Apple'
      },
      slots: {
        default: `
          <option value="apple">Apple</option>
          <option value="orange">Orange</option>
          <option value="lemon">Lemon</option>
        `
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when select', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.findAll('option').at(1).element.selected = true;
      wrapper.find('select').trigger('input');
    });

    it('emit input', () => {
      expect(wrapper.emitted('input')[0]).toEqual(['orange']);
    });
  });
});
