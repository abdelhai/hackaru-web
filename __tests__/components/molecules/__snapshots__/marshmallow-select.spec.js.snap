// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarshmallowSelect render correctly 1`] = `
<div
  class="marshmallow-select"
>
  <btn-stub
    class="has-dropshadow"
  >
    
    Apple
  
  </btn-stub>
   
  <select
    aria-label="Fruits"
  >
    <option
      value="apple"
    >
      Apple
    </option>
     
    <option
      value="orange"
    >
      Orange
    </option>
     
    <option
      value="lemon"
    >
      Lemon
    </option>
  </select>
</div>
`;
